File: cop3530/assignment2_src/two_largest.cc
#include <iostream>
#include <vector>

void TwoLargest(const std::vector<int>& A) {
    int large_1 = 0;
    int large_2 = 0;

    for (int i = 0; i < A.size(); ++i) {
        if (A[i] > large_1) {
            large_2 = large_1;
            large_1 = A[i];
        } else if (A[i] > large_2) {
            large_2 = A[i];
        }
    }

    std::cout << "The two largest numbers are: " << large_1 << " and " << large_2 << std::endl;
}

int main() {
    int n;
    std::cout << "Enter the size of the list: ";
    std::cin >> n;

    std::vector<int> A(n);
    std::cout << "Enter the elements of the list: ";
    for (int i = 0; i < n; ++i) {
        std::cin >> A[i];
    }

    TwoLargest(A);

    return 0;
}

File: cop3530/assignment2_src/two_largest.py
# Function to find two largest numbers in a list

def two_largest(arr):
    large_1 = float('-inf')  # Set initial value to negative infinity
    large_2 = float('-inf')
    
    for num in arr:
        if num > large_1:
            large_2 = large_1
            large_1 = num
        elif num > large_2:
            large_2 = num
    
    return large_1, large_2

# Get user input for the list
def get_user_input():
    arr = list(map(int, input("Enter numbers separated by spaces: ").split()))
    return arr

# Main execution
if __name__ == "__main__":
    # Get user input
    arr = get_user_input()
    
    # Find two largest numbers
    large_1, large_2 = two_largest(arr)
    
    # Output the result
    print(f"The two largest numbers are: {large_1} and {large_2}")
